www/source/blog/2018-08-18-lift-and-shift-hab.html.md

```diff
@@ -0,0 +1,396 @@
+
+Let's concentrate for now on creating deployment artifacts with Habitat.  Below we use the Studio to export different package formats:
+```bash
+$ cd table-setting
+$ hab studio enter
+[STUDIO] build
+[STUDIO] hab pkg export tar $(ls -1t results/*.hart | head -1)
+[STUDIO] hab pkg export docker $(ls -1t results/*.hart | head -1)
+```
+
+We also use the Habitat maintained `core/mysql` image for some deployments, let's install and export that locally also:
+
+```bash
+[STUDIO] hab pkg install core/mysql
+» Installing core/mysql
+☁ Determining latest version of core/mysql in the 'stable' channel
+☛ Verifying core/mysql/5.7.21/20180609181124
+... output curtailed
+★ Install of  core/mysql/5.7.21/20180609181124 complete with 25 new packages installed.
+```
+
+Note the package version from the installation command, we will now use this to export a Docker image locally:
+
+```
+[STUDIO] hab pkg export docker core/mysql/5.7.21/20180609181124
+```
+
+For reference, all of the plans built and maintained by the Habitat Core Team are [available here](https://github.com/habitat-sh/core-plans).
+
+The resulting artifacts are all we need for the deployment scenarios below.
+
+## Bare metal or standalone VM deployment
+
+Let's assume we want to deploy Table Setting to an Ubuntu machine and use a file-based SQLite database backend.  In a real-world setup the versioned `tar` deployment file might be stored in a dedicated artifact repository but we will assume it is locally available on the target machine here.
+
+Relying on the Habitat package naming convention, we can run the package as follows:
+
+```bash
+$ cat hab_tar_deployment_ubuntu.sh
+#!/bin/bash
+# one-time setup - packages can specify a linux user to run under, by default this is the hab user 
+sudo adduser --disabled-password --gecos "" hab
+# assumes local package file
+sudo tar zxvf *-table-setting*.tar.gz -C /
+# run app in SQLite file mode
+export DB_TYPE=sqlite_file
+# install and run with the habitat supervisor
+sudo /hab/bin/hab sup run &
+sudo /hab/bin/hab svc load habskp/table-setting
+```
+
+Running this on a locally running Ubuntu VM looks as follows:
+
+![Running on_ubuntu_vm](media/2018-08-18-lift-and-shift-hab/run_tar_ubuntu_vm.png)
+
+
+This is arguably simpler than the steps in the previous section.  We've already successfully decoupled a lot of the package internals from the deployment process.  Although some knowledge is required to run the `tar` Habitat package locally, this would be a similar set of steps for *any* Habitat package.  Now let's see what else we can do. 
+
+## Running with Docker Compose
+
+For this case, let's take a look at the Docker Compose manifest in the root of the Table Setting repository:
+
+```bash
+$ cat docker-compose.yml
+version: '3'
+services:
+    mysql:
+        image: core/mysql
+        environment:
+          HAB_MYSQL: |
```

Comment:
Long-term, I'm not sure that if we'll want to continue to support configuration of Habitat services by environment variables (e.g. `HAB_MYSQL`) like this, because the usage is very fragile and confusing; see [this discussion thread](https://github.com/habitat-sh/habitat/issues/4042#issuecomment-404595963) for additional details. I generally recommend not using them at all.

Mounting a `user.toml` file into the container is another viable option, though it may complicate your example.